web/components/contract/private-contract.tsx

```diff
@@ -0,0 +1,177 @@
+import { getInitialProbability } from 'common/calculate'
+import {
+  AnyContractType,
+  BinaryContract,
+  Contract,
+  PseudoNumericContract,
+} from 'common/contract'
+import { ContractMetric } from 'common/contract-metric'
+import { getTotalContractMetrics } from 'common/supabase/contract-metrics'
+import { removeUndefinedProps } from 'common/util/object'
+import { useEffect, useState } from 'react'
+import { useBetCount, useBets } from 'web/hooks/use-bets-supabase'
+import { useComments } from 'web/hooks/use-comments-supabase'
+import { usePrivateContract } from 'web/hooks/use-contracts'
+import { getInitialRelatedMarkets } from 'web/hooks/use-related-contracts'
+import { useUserById } from 'web/hooks/use-user-supabase'
+import {
+  getBinaryContractUserContractMetrics,
+  getTopContractMetrics,
+} from 'web/lib/firebase/contract-metrics'
+import { db } from 'web/lib/supabase/db'
+import {
+  ContractPageContent,
+  ContractParams,
+} from 'web/pages/[username]/[contractSlug]'
+import {
+  InaccessiblePrivateThing,
+  LoadingPrivateThing,
+} from '../groups/private-group'
+import {
+  getBetPoints,
+  getHistoryDataBets,
+  getUseBetLimit,
+  shouldUseBetPoints,
+} from './contract-page-helpers'
+
+export function PrivateContractPage(props: { contractSlug: string }) {
+  const { contractSlug } = props
+  const contract = usePrivateContract(contractSlug, 1000)
+
+  if (contract === undefined) {
+    return <LoadingPrivateThing />
+  } else if (contract === null)
+    return <InaccessiblePrivateThing thing="market" />
+  else {
+    return <ContractParamsPageContent contract={contract} />
+  }
+}
+
+export function ContractParamsPageContent(props: {
+  contract: Contract<AnyContractType>
+}) {
+  const { contract } = props
+  const contractParams = useContractParams(contract)
+  return <ContractPageContent contractParams={contractParams} />
+}
+
+export function useContractParams(contract: Contract) {
```

Comment:
Hmm, this hook has a bunch of sub hooks that each make a new query. And each query sets state, triggering a ton of extra renders that slow things down.

An alternative would be to create an api endpoint that sends all the contract params data in a json blob, if you are authenticated and have permission.
I guess if RLS is working, it could just be a regular function that fetches all this data. And then you could call it in the `getStaticProps` for public markets as well.